workflow-ui/core-android/src/main/java/com/squareup/workflow1/ui/ViewRegistry.kt

```diff
@@ -130,8 +130,7 @@ public fun <RenderingT : Any>
   ViewRegistry.getFactoryForRendering(rendering: RenderingT): ViewFactory<RenderingT> {
   @Suppress("UNCHECKED_CAST")
   return getFactoryFor(rendering::class)
-    ?: (rendering as? AndroidViewRendering<*>)?.viewFactory as? ViewFactory<RenderingT>
-    ?: (rendering as? Named<*>)?.let { NamedViewFactory as ViewFactory<RenderingT> }
+    ?: (unwrap(rendering) as? AndroidViewRendering<*>)?.viewFactory as? ViewFactory<RenderingT>
```

Comment:
This is tricky, and I think wrong so far. A lot of wrappers exist precisely because they want to change how the view is built, like the BackButtonScreen touched by this PR. If a `BackButtonScreen` gets wrapped by something else, and in turn wraps and `AndroidViewRendering`, this would skip `BackButtonViewFactory` entirely.

This needs to be more visitor-like. Have to be careful not to over engineer it.

Reply:
Maybe I'm over thinking this. `BackStackScreen` shouldn't implement `Wrapper` precisely because it doesn't entirely delegate UI work. And so we should unwrap `rendering` before calling `getFactoryFor`.

Reply:
Yeah, that makes sense — on a non-Android UI trying to use `BackButtonScreen`, we would get a failed binding error, exactly the right thing to remind us that we need a non-Android implementation.